sweep/sweeper.go

```diff
@@ -833,6 +834,103 @@ func (s *UtxoSweeper) clusterBySweepFeeRate(inputs pendingInputs) []inputCluster
 	return inputClusters
 }
 
+// zipClusters merges pairwise clusters from as and bs such that cluster a from
+// as is merged with a cluster from bs that has at least the fee rate of a.
+// This to ensure we don't delay confirmation by decreasing the fee rate (the
+// lock time inputs are typically second level HTLC transactions, that are time
+// sensitive).
+func zipClusters(as, bs []inputCluster) []inputCluster {
+	// Sort the clusters by decreasing fee rates.
+	sort.Slice(as, func(i, j int) bool {
+		return as[i].sweepFeeRate >
+			as[j].sweepFeeRate
+	})
+	sort.Slice(bs, func(i, j int) bool {
+		return bs[i].sweepFeeRate >
+			bs[j].sweepFeeRate
+	})
+
+	var (
+		finalClusters []inputCluster
+		j             int
+	)
+
+	// Go through each cluster in as, and merge with the next one from bs
+	// if it has at least the fee rate needed.
+	for i := range as {
+		a := as[i]
+
+		// Check the next one from bs if any.
+		if j < len(bs) {
+			b := bs[j]
+
+			// If the fee rate is less than a's, we break. All the
+			// remining clusters from bs have lower fee rate.
+			if b.sweepFeeRate < a.sweepFeeRate {
+				break
+			}
+
+			// We can merge.
+			merged := mergeClusters(a, bs[j])
+			finalClusters = append(finalClusters, merged...)
```

Comment:
what if there are multiple clusters in b that an be merged with _this_ a?

Reply:
Then it will be merged with the first one compatible. This is definitely not an optimized version, there are certainly "smarter" ways of finding clusters to merge with, but wanted to keep it simple for now.